### PR TITLE
Log clusterbus handshake timeout failures

### PR DESCRIPTION
**Pull request by bentotten**
This adds a log when a handshake fails for a timeout. This can help troubleshoot cluster asymmetry issues caused by failed MEETs

### CODE REVIEW

**Comment by bentotten** [comment]
Odd - will check failures shortly, passed in local fork

**Comment by zuiderkwast** [comment]
> Odd - will check failures shortly, passed in local fork

I've seen those test cases fail before. I restarted the failed jobs.

**Review by enjoy-binbin — APPROVED** [review]
This seems ok to me. Could it be that there is a bad node that causes this log to be printed continuously?

**Comment by bentotten** [comment]
> This seems ok to me. Could it be that there is a bad node that causes this log to be printed continuously?

Do you mean in situations where the MEET is resent?

File: src/cluster_legacy.c [hunk 1]
```diff
@@ -4872,6 +4872,8 @@ static int clusterNodeCronHandleReconnect(clusterNode *node, mstime_t handshake_
     /* A Node in HANDSHAKE state has a limited lifespan equal to the
      * configured node timeout. */
     if (nodeInHandshake(node) && now - node->ctime > handshake_timeout) {
+        serverLog(LL_WARNING, "Clusterbus handshake timeout [%s]:%d after a timeout of %lld", node->ip,
+                  node->cport, handshake_timeout);
```
Comment (madolson) [inline src/cluster_legacy.c]:
```suggestion
        serverLog(LL_WARNING, "Clusterbus handshake timeout %s:%d after %lldms", node->ip,
                  node->cport, handshake_timeout);
```
Minor suggestion to indicate clearly the timeout is in ms and remove the square bracket. There is some usage of both with and without the square bracket.

Comment (bentotten) [reply src/cluster_legacy.c]: Complete

**Comment by enjoy-binbin** [comment]
> > This seems ok to me. Could it be that there is a bad node that causes this log to be printed continuously?

> Do you mean in situations where the MEET is resent?

Yes, there should only be one print in a handshake_timeout, so it should be fine i guess.

**Comment by bentotten** [comment]
> > > This seems ok to me. Could it be that there is a bad node that causes this log to be printed continuously?
> 
> > Do you mean in situations where the MEET is resent?
> 
> Yes, there should only be one print in a handshake_timeout, so it should be fine i guess.

Sounds great - is this able to be merged?

**Review by madolson — APPROVED** [review]

**Pull request merged**